pkg/provider/authorization.go

```diff
@@ -0,0 +1,65 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/lambda"
+)
+
+type Authorization struct {
+	Name     string `json:"name"`
+	Password string `json:"password"`
+}
+
+func CheckAuthorization(user, password, authorizationLambda string, jwtInfo *JWT) (*JWT, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(jwtInfo.Region),
+		Credentials: credentials.NewStaticCredentials(jwtInfo.AWSAccessKeyID, jwtInfo.AWSSecretAccessKey, ""),
+	})
+	if err != nil {
+		return nil, err
+	}
+	svc := lambda.New(sess)
+	payload, _ := json.Marshal(Authorization{
+		Name:     user,
+		Password: password,
+	})
+
+	input := &lambda.InvokeInput{
+		FunctionName: aws.String(authorizationLambda),
+		Payload:      payload,
+	}
+
+	result, err := svc.Invoke(input)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	} else {
+		response := string(result.Payload)
+		if response == "\"OK\"" {
+			origJSON, err := json.Marshal(&jwtInfo)
+			if err != nil {
+				return nil, err
+			}
+
+			userJWT := JWT{}
+			if err = json.Unmarshal(origJSON, &userJWT); err != nil {
+				return nil, err
+			}
```

Comment:
I see why you would not want to modify `origJSON` but there's no need to marshal and unmarshall it again. You could directly copy the string fields of the struct. I would also suggest to rename `origJSON` as it is not a JSON.

Reply:
Yeah, I took that code from some web page I found. Basically, I wanted to be able to copy the struct without having to hardcode individual fields. Naming could be better, I agree.